### PR TITLE
[JUJU-1631] Rewrite api/client/block unit tests to use gomock

### PR DESCRIPTION
**Pull request by cderici**
The unit tests for `api/client/block` now use go mock and not juju/testing. All unit tests should pass.

### CODE REVIEW

File: api/client/block/client_test.go [hunk 1]
```diff
@@ -4,135 +4,134 @@
...
 type blockMockSuite struct {
-	coretesting.BaseSuite
-	blockClient *block.Client
 }
 
 var _ = gc.Suite(&blockMockSuite{})
 
 func (s *blockMockSuite) TestSwitchBlockOn(c *gc.C) {
+	ctrl := gomock.NewController(c)
 	called := false
```
Comment (hmlanigan) [inline api/client/block/client_test.go:26]: "called" is no required. It's implicit with the go mock EXPECT() line.

Comment (cderici) [reply api/client/block/client_test.go:26]: Yeah that's why I was doing a `DoAndReturn`, good call, thanks! :+1:

**Review by hmlanigan — APPROVED** [review]
LGTM

**Pull request merged**